src/api/entities/SecurityToken/index.ts

```diff
@@ -283,4 +285,14 @@ export class SecurityToken extends Entity<UniqueIdentifiers> {
 
     return null;
   }
+
+  /**
+   * Sets a treasury DID for this token
+   *
+   * @param args.target - identity to set as treasury. Empty to reset it
+   */
+  public setTreasury(args: SetTreasuryDidParams): Promise<TransactionQueue<void>> {
+    const { ticker } = this;
+    return setTreasuryDid.prepare({ ticker, ...args }, this.context);
+  }
```

Comment:
Instead of having a separate procedure and endpoint, this should be part of modifyToken and the modify endpoint